lib/base/process.hpp

```diff
@@ -25,7 +26,7 @@ struct ProcessResult
 	pid_t PID;
 	double ExecutionStart;
 	double ExecutionEnd;
-	long ExitStatus;
+	int_fast64_t ExitStatus;
```

Comment:
I honestly don't understand why you are changing this, why not keep using `long`? What benefits does this fancy type have over the original type in this specific context and its intended usage? I would simply change the type of the `CheckResult#exit_status` attribute from `int` to `long` instead.

Reply:
> I'm addressing a 32-bit overflow here, so I need 64+ bits.

First of all, it is not `ProcessResult#ExitStatus` that's overflowing `int32`, but `CheckResult#exit_status`, which has been of type `int` so far. Secondly, `int_fast64_t` is just an alias of some other `64` bit types, e.g. on my Mac it is just an alias of `int64_t` and the compiler might do some optimisations based on this type when doing some sophisticated arithmetic, but it is not intended to overcome `64+` overflows.

I just don't think it's needed here and `long` should be way more than enough for signals/exit codes IMHO, given that there are no 64+ systems at all.

Reply:
1. CheckResult#exit_status gets set from ProcessResult#ExitStatus, so both need a large type
2. Just grep for int_fast in the code - if we need X bits, we declare it explicitly
3. long has 32 bits, even on x64 Windows https://learn.microsoft.com/en-us/cpp/build/common-visual-cpp-64-bit-migration-issues?view=msvc-170&redirectedfrom=MSDN